extension/entitystore/extension.go

```diff
@@ -0,0 +1,257 @@
+
+type serviceProviderInterface interface {
+	startServiceProvider()
+	addEntryForLogFile(LogFileGlob, ServiceAttribute)
+	addEntryForLogGroup(LogGroupName, ServiceAttribute)
+	logFileServiceAttribute(LogFileGlob, LogGroupName) ServiceAttribute
+	getServiceNameAndSource() (string, string)
+}
+
+type EntityStore struct {
+	logger *zap.Logger
+	config *Config
+	done   chan struct{}
+
+	// mode should be EC2, ECS, EKS, and K8S
+	mode string
+
+	kubernetesMode string
+
+	// ec2Info stores information about EC2 instances such as instance ID and
+	// auto scaling groups
+	ec2Info EC2Info
+
+	// eksInfo stores information about EKS such as pod to service Env map
+	eksInfo *eksInfo
+
+	// serviceprovider stores information about possible service names
+	// that we can attach to the entity
+	serviceprovider serviceProviderInterface
+
+	// nativeCredential stores the credential config for agent's native
+	// component such as LogAgent
+	nativeCredential client.ConfigProvider
+
+	metadataprovider ec2metadataprovider.MetadataProvider
+
+	stsClient stsiface.STSAPI
+
+	podTerminationCheckInterval time.Duration
+}
+
+var _ extension.Extension = (*EntityStore)(nil)
+
+func (e *EntityStore) Start(ctx context.Context, host component.Host) error {
+	// Get IMDS client and EC2 API client which requires region for authentication
+	// These will be passed down to any object that requires access to IMDS or EC2
+	// API client so we have single source of truth for credential
+	e.done = make(chan struct{})
+	e.metadataprovider = getMetaDataProvider()
+	e.mode = e.config.Mode
+	e.kubernetesMode = e.config.KubernetesMode
+	e.podTerminationCheckInterval = podTerminationCheckInterval
+	ec2CredentialConfig := &configaws.CredentialConfig{
+		Profile:  e.config.Profile,
+		Filename: e.config.Filename,
+	}
+	switch e.mode {
+	case config.ModeEC2:
+		e.ec2Info = *newEC2Info(e.metadataprovider, e.done, e.config.Region, e.logger)
+		go e.ec2Info.initEc2Info()
+	}
+	if e.kubernetesMode != "" {
+		e.eksInfo = newEKSInfo(e.logger)
+		// Starting the ttl cache will automatically evict all expired pods from the map
+		go e.StartPodToServiceEnvironmentMappingTtlCache(e.done)
+	}
+	e.serviceprovider = newServiceProvider(e.mode, e.config.Region, &e.ec2Info, e.metadataprovider, getEC2Provider, ec2CredentialConfig, e.done, e.logger)
+	go e.serviceprovider.startServiceProvider()
+	return nil
+}
+
+func (e *EntityStore) Shutdown(_ context.Context) error {
+	close(e.done)
+	return nil
+}
+
+func (e *EntityStore) Mode() string {
+	return e.mode
+}
+
+func (e *EntityStore) KubernetesMode() string {
+	return e.kubernetesMode
+}
+
+func (e *EntityStore) EKSInfo() *eksInfo {
+	return e.eksInfo
+}
+
+func (e *EntityStore) EC2Info() EC2Info {
+	return e.ec2Info
+}
+
+func (e *EntityStore) SetNativeCredential(client client.ConfigProvider) {
+	e.nativeCredential = client
+}
+
+func (e *EntityStore) NativeCredentialExists() bool {
+	return e.nativeCredential != nil
+}
+
```

Comment:
It doesn't look like these are ever used.

Reply:
Add a commit to get rid of additional code

Reply:
Non-blocking comment.